Sources/Extensions/Foundation/UserDefaultsExtensions.swift

```diff
@@ -38,4 +38,27 @@ public extension UserDefaults {
 		return object(forKey: key) as? Date
 	}
 	
+    /// SwifterSwift: Retrieves a Codable object from UserDefaults.
+    ///
+    /// - Parameters:
+    ///   - type: Class that conforms to the Codable protocol.
+    ///   - key: key to find the Codable object for.
+    /// - Returns: Codable object for key (if exists).
+    public func object<T: Codable>(_ type: T.Type, with key: String) -> T? {
```

Comment:
Lets add a decoder argument to this function incase user wanted to use a custom decoder:
```swift
public func object<T: Codable>(_ type: T.Type, with key: String, usingDecoder decoder: JSONDecoder = JSONDecoder()) -> T? {
    guard let data = self.value(forKey: key) as? Data else { return nil }
    return try? decoder.decode(type.self, from: data)
}
```

Same here let's add an encoder argument, incase user wanted to use a custom encoder:

```swift
public func set<T: Codable>(object: T, forKey key: String, usingEncoder encoder: JSONEncoder = JSONEncoder()) {
    let data = try? encoder.encode(object)
    self.set(data, forKey: key)
}
```

Reply:
I see. Will update, thanks!